src/pages/OnboardingWork/BaseOnboardingWork.tsx

```diff
@@ -0,0 +1,153 @@
+            // code units.
+            ErrorUtils.addErrorMessage(errors, 'work', ['common.error.characterLimitExceedCounter', {length: [...work].length, limit: CONST.TITLE_CHARACTER_LIMIT}]);
+        }
+
+        return errors;
+    };
+
+    const handleGoBack = useCallback(() => {
+        Navigation.goBack();
+    }, []);
+
+    const WorkFooterInstance = <OfflineIndicator />;
+
+    return (
+        <View style={[styles.h100, styles.defaultModalContainer, shouldUseNativeStyles && styles.pt8]}>
+            <HeaderWithBackButton
+                shouldShowBackButton
+                progressBarPercentage={75}
+                onBackButtonPress={handleGoBack}
+            />
+            <KeyboardAvoidingView
+                style={[styles.flex1, styles.dFlex]}
+                behavior="padding"
+            >
+                <FormProvider
+                    style={[styles.flexGrow1, shouldUseNarrowLayout && styles.mt5, styles.mb5, shouldUseNarrowLayout ? styles.mh8 : styles.mh5]}
+                    formID={ONYXKEYS.FORMS.ONBOARDING_PERSONAL_WORK}
+                    footerContent={isSmallScreenWidth && WorkFooterInstance}
+                    validate={validate}
+                    onSubmit={completeEngagement}
+                    submitButtonText={translate('common.continue')}
+                    enabledWhenOffline
+                    submitFlexEnabled
+                    shouldValidateOnBlur
+                    shouldValidateOnChange
+                    shouldTrimValues={false}
+                >
+                    <View style={[shouldUseNarrowLayout ? styles.flexRow : styles.flexColumn, styles.mb5]}>
+                        <Text style={[styles.textHeadlineH1, styles.textXXLarge]}>{translate('onboarding.whereYouWork')}</Text>
+                    </View>
+                    <View style={styles.mb4}>
+                        <InputWrapper
+                            InputComponent={TextInput}
+                            inputID={INPUT_IDS.WORK}
+                            name="fwork"
+                            label={translate('common.businessName')}
+                            aria-label={translate('common.businessName')}
+                            role={CONST.ROLE.PRESENTATION}
+                            shouldSaveDraft
+                            maxLength={CONST.TITLE_CHARACTER_LIMIT}
+                            spellCheck={false}
+                            autoFocus
+                        />
+                    </View>
+                </FormProvider>
+            </KeyboardAvoidingView>
+        </View>
+    );
+}
+
+BaseOnboardingWork.displayName = 'BaseOnboardingWork';
+
+export default withCurrentUserPersonalDetails(
+    withOnyx<BaseOnboardingWorkProps, BaseOnboardingWorkOnyxProps>({
+        onboardingPurposeSelected: {
+            key: ONYXKEYS.ONBOARDING_PURPOSE_SELECTED,
+        },
```

Comment:
Do you want to use useOnyx for this?

Reply:
Same, this change will requires more testing from my side, could we leave if for the future?